plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/NewApplicationWizard.java

```diff
@@ -220,14 +232,37 @@ public void done(IJobChangeEvent event) {
 						//No need to import the project from git, it's already here
 						return;
 					}
-					final Map<IProject, Collection<IBuildConfig>> projectsAndBuildConfigs = getBuildConfigs(createJob.getResources());
+					Collection<IResource> resources = createJob.getResources();
+					final Map<IProject, Collection<IBuildConfig>> projectsAndBuildConfigs = getBuildConfigs(resources);
 					if (projectsAndBuildConfigs.isEmpty()) {
 						return;
 					}
+					Connection connection = model.getConnection();
 					Display.getDefault().asyncExec(new Runnable() {
 						@Override
 						public void run() {
 							ImportApplicationWizard wizard = new ImportApplicationWizard(projectsAndBuildConfigs);
+							wizard.addImportJobChangeListener(new JobChangeAdapter() {
+								
+								@Override
+								public void done(IJobChangeEvent event) {
+									IService service = getService(resources);
+									List<org.eclipse.core.resources.IProject> importedProjects = wizard.getImportJob().getImportedProjects();
+									if (service != null && importedProjects.size() == 1) {
+										Display.getDefault().asyncExec(new Runnable() {
+
+								            @Override
+								            public void run() {
+								            	if (MessageDialog.openQuestion(getShell(), "Create server adapter", NLS.bind(
```

Comment:
I dont think that we should ask the user. I think that we should provide a wizard page (like we do in v2) that holds a checkbox ([x] Create Server adapter for easy publishing).
Possibly we even should have the server adapter wizard page included in the new application wizard